lib/jsdom/living/websocket/WebSocket.webidl

```diff
@@ -0,0 +1,31 @@
+// https://html.spec.whatwg.org/multipage/web-sockets.html#the-websocket-interface
+
+enum BinaryType { "blob", "arraybuffer" };
+[Constructor(USVString url, optional (DOMString or sequence<DOMString>) protocols = []), Exposed=(Window,Worker)]
```

Comment:
It looks like this is going to have to contain `WebIDL2JSFactory` attribute as well, since EventTarget won't function w/o a `ownerDocument`, which is per-window.

Basically, instead of having a shared `WebSocket` interface in `living/index.js`, you would want to create a new interface in the Window constructor. Check out f77cc1e3a9eb38fb7db6affa5a9c3d2acca1b351 and https://github.com/jsdom/webidl2js#webidl2jsfactory.